### PR TITLE
Added concepts for react and angular2. wrote a bit about speed benchmark

### PR DESCRIPTION
**Pull request by DanielLChang**
Fixing issue #5

### CODE REVIEW

File: AngularVReact.md [hunk 1]
```diff
@@ -1,5 +1,26 @@
 # Introduction
-Comparison points on Angular vs. React. Angular is full MVC while React is simply V.
+Before we get started, **I am in no way saying one framework is better than the other.** These are just some comparison points on Angular vs. React. In the end it all comes down to whatever you're looking for.
```
Comment (davidhu2000) [inline AngularVReact.md:2]: This sentence feels unnecessary. What if we just remove all the winner comments?

File: AngularVReact.md [hunk 2]
```diff
@@ -1,5 +1,26 @@
...
+2. As we don't interact directly with the DOM, a browser is not needed to test React
+3. The virtual DOM has the ability to connect to other entities (Native and Electron).
+
+- State: Components in React have a state which represents component-related data. Updating state will allow the page to be reactive.
+
+**Angular:**
+
+- Watchers: Watchers (similar to event listeners) are attached to each component and each time a component is changed, watchers check if we should modify something else; and if needed, make appropriate modifications. This part in Angular 2 is way fast than its previous version so each time a component is changed, we don’t have to run any verifications on objects (depending on immutable elements).
+
+- Typescript: Angular2 requires Typescript. Compared to vanillaJS, Typescript offers better code organization, typing, and annotations so Typescript is better from a 'strictness' perspective but in regards to 'learning' you will often find yourself learning Typescript and Angular at the same time.
```
Comment (davidhu2000) [inline AngularVReact.md]: I don't believe this is actually true. Angular 2 comes default with Typescript, but you can write in vanilla javascript if you so choose. 

see: [here](https://daveceddia.com/angular-2-in-plain-js/)

File: Javascript.md [hunk 3]
```diff
@@ -1,5 +0,0 @@
-# JavaScript: Basics and Tricky Questions
-
-### What are the differences between null and undefined?
-- `null`: No value. Null can be assigned to any variable but is not an object. typeof null returns `object`
-- `undefined`
```
Comment (davidhu2000) [inline Javascript.md:5]: why is this removed? This feels like a pretty section to keep, maybe just elaborate more?

**Review by davidhu2000 — CHANGES_REQUESTED** [review]
Please see comments.

File: AngularVReact.md (hunk 1 above)
Comment (DanielLChang) [reply AngularVReact.md:2]: oh yeah that works LOL

File: Javascript.md (hunk 3 above)
Comment (DanielLChang) [reply Javascript.md:5]: I didnt realize it was already covered in the frontend doc haha.

Comment (davidhu2000) [reply Javascript.md:5]: would it make more sense to migrate topics like this from `frontend.md` to `javascript.md`?

Comment (DanielLChang) [reply Javascript.md:5]: hmm, maybe. like have a separate doc for js and css?

File: AngularVReact.md [hunk 4]
```diff
@@ -20,42 +20,32 @@ Angular is full MVC while React is simply V.
 
 - Watchers: Watchers (similar to event listeners) are attached to each component and each time a component is changed, watchers check if we should modify something else; and if needed, make appropriate modifications. This part in Angular 2 is way fast than its previous version so each time a component is changed, we don’t have to run any verifications on objects (depending on immutable elements).
 
-- Typescript: Angular2 requires Typescript. Compared to vanillaJS, Typescript offers better code organization, typing, and annotations so Typescript is better from a 'strictness' perspective but in regards to 'learning' you will often find yourself learning Typescript and Angular at the same time.
-
```
Comment (davidhu2000) [inline AngularVReact.md]: I think we should keep this sentence, but instead of saying "Angular2 requires Typescript`, we can change it to "Angular2 comes default with Typescript". I think it is beneficial to include some of the benefits of Typescript over vanilla jS.

**Review by davidhu2000 — CHANGES_REQUESTED** [review]
See comment.

File: Javascript.md (hunk 3 above)
Comment (davidhu2000) [reply Javascript.md:5]: it might be a good idea to separate them, otherwise the frontend file might get too long.

Comment (DanielLChang) [reply Javascript.md:5]: yeah okay. i'll probably try to split it up tomorrow or something. too lazy now HAHA

**Review by davidhu2000 — APPROVED** [review]
